test/e2e/suite/test_node.go

```diff
@@ -0,0 +1,117 @@
+package suite
+
+import (
+	"context"
+	"path/filepath"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"github.com/aws/eks-hybrid/test/e2e"
+	"github.com/aws/eks-hybrid/test/e2e/constants"
+	"github.com/aws/eks-hybrid/test/e2e/credentials"
+	"github.com/aws/eks-hybrid/test/e2e/ec2"
+	"github.com/aws/eks-hybrid/test/e2e/kubernetes"
+	"github.com/aws/eks-hybrid/test/e2e/peered"
+)
+
+type TestNode struct {
+	flakyCode    *FlakyCode
+	instanceName string
+	k8sVersion   string
+	node         *peered.PeerdNode
+	nodeName     string
+	os           e2e.NodeadmOS
+	provider     e2e.NodeadmCredentialsProvider
+	serialOutput peered.ItBlockCloser
+	test         *peeredVPCTest
```

Comment:
is there a way to avoid having this in the test node struct?
I don't have anything concrete but it feels weird. Is this because we use a lot of the fields from test